### ISSUE TITLE
unittest failure `test_force_constants_self_consistency` in `Si_test.py`

### ISSUE THREAD

**Issue by bjmorgan** [abelcarreras/DynaPhoPy#4]
If I run the test script `Si_test.py` I get an error
`ValueError: operands could not be broadcast together with shapes` in the test `test_force_constants_self_consistency`.

https://github.com/abelcarreras/DynaPhoPy/blob/f4e16b9a1068ac4dd5d571252c28e95ac2d423af/unittest/Si_test.py

```python
def test_force_constants_self_consistency(self):
    self.calculation.select_power_spectra_algorithm(2)
    renormalized_force_constants = self.calculation.get_renormalized_force_constants().get_array()
    harmonic_force_constants = self.calculation.dynamic.structure.get_force_constants().get_array()

    self.assertEqual(np.allclose(renormalized_force_constants, harmonic_force_constants, rtol=1, atol=1.e-2), True)
```

The issue appears to be that `renormalized_force_constants` and `harmonic_force_constants` have different shapes. In this case these array shapes are (2,64,3,3) and (64,64,3,3), so `np.allclose()` raises a `ValueError`.

**Comment by abelcarreras** [comment]
Sorry for late reply, I was traveling these days,

That's weird! In my computer this test works and Travis CI didn't complain (https://travis-ci.org/abelcarreras/DynaPhoPy/branches)
Which OS are you using? Are you using Master branch?

**Comment by bjmorgan** [comment]
Linux, using the `master` branch.

> Travis CI didn't complain

I noticed that the previous commit to master passed on Travis. But my pull request https://github.com/abelcarreras/DynaPhoPy/pull/3 fails on Travis, and it appears as though your latest commit to the `development` branch now fails on the same tests.

**Comment by abelcarreras** [comment]
Yeah, that's what I am checking right now.
I'm working on this problem.

**Comment by abelcarreras** [comment]
I think I found it!
Seems that something has changed in phonopy recently. That's why TravisCI didn't fail before
and now it's failing. Probably I will have a fix very soon.

**Comment by abelcarreras** [comment]
It seems that a bug has been introduced in phonopy v1.12.6.32 in dynmat2fc function which is critical for dynaphopy. I wrote an issue in phonopy github and I hope it will be fixed soon.
By now the solution would be to downgrade phonopy to 1.12.6.26, I will modify the requirements.txt file
to require this version specific version of phonopy.

**Comment by abelcarreras** [comment]
I talked with phonopy developer and he told me that this is not a bug, is a change of the dynmat2fc default behavior. It seems that the latest version of phonopy can work with a compact format of the force constants. This is now the ouput format of dynmat2fc function which is not compatible with dynaphopy unitttests.

However, since dynaphopy uses many phonopy functions to obtain the eigenvectors & frequencies as well as to read and write force constants this change does not affect the normal usage of dynaphopy (only tests). Only that using a new version of phonopy the renormalized  obtained will be in compact format (which are totally compatible with phonopy).

According to phonopy developer additional functionality will be added to dynmat2fc to choose between compact or full size force constants format. Now this is in development branch but soon will be released in stable version.

Therefore, in order to keep back compatibility for now I have decided to wait until the new functionality is added to dynmat2fc since latest version of phonopy is still fully compatible with dynaphopy (except for unittests) and modify the requirement to install phonopy <v1.12.6.26 (which is the latests compatible with unittests) so TravisCI don't fail. When the new functionality is added to stable release (i.e. added to PyPI & conda) then I will modify dynaphopy code to support this new dynmat2fc functionality.

Will this work for you?
